CMakeLists.txt

```diff
@@ -55,11 +53,17 @@ option(DLAF_BUILD_DOC "Build documentation" OFF)
 option(DLAF_WITH_PRECOMPILED_HEADERS "Use precompiled headers." OFF)
 option(DLAF_WITH_SCALAPACK "Build ScaLAPACK-like C API (requires ScaLAPACK)" OFF)
 
-if(DLAF_WITH_MKL AND DLAF_WITH_MKL_LEGACY)
-  message(SEND_ERROR "DLAF_WITH_MKL and DLAF_WITH_MKL_LEGACY are mutually exclusive.")
-endif()
+try_compile(
+  DLAF_WITH_MKL_TRY_COMPILE "${PROJECT_SOURCE_DIR}/cmake/tests"
+  "${PROJECT_SOURCE_DIR}/cmake/tests/mkl_set_num_threads.cpp"
+  CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${LAPACK_INCLUDE_DIR}"
+  LINK_LIBRARIES ${LAPACK_LIBRARY}
+)
+# TODO: Maybe don't need to print this? Though maybe it's nice to have...
+message(WARNING "DLAF_WITH_MKL_TRY_COMPILE=${DLAF_WITH_MKL_TRY_COMPILE}")
+option(DLAF_WITH_MKL "Enable oneMKL as provider for BLAS and LAPACK" "${DLAF_WITH_MKL_TRY_COMPILE}")
```

Comment:
To do: the description is outdated (this option applies to all MKL versions now).